### PR TITLE
Use Babel to build React

### PR DESCRIPTION
**Pull request by spicyj**
Size comparison:

```
   raw     gz Compared to master @ 6ed98ec0c83918d91b16517544e658e2ec068070
     =      = build/JSXTransformer.js
-15736  -3247 build/react-with-addons.js
  +287     +7 build/react-with-addons.min.js
-14412  -2887 build/react.js
  +274    +15 build/react.min.js
```

Differences mostly look to be various bits of whitespace that Babel ends up removing during its transforms. In minified files, mostly additions of `"use strict";`.

### CODE REVIEW

**Comment by vipulnsward** [comment]
:+1:

**Comment by spicyj** [comment]
(cc @sebmck)

File: src/test/mocks.js [hunk 1]
```diff
@@ -69,7 +69,7 @@ function makeComponent(metadata) {
 
         instances.push(this);
         calls.push(Array.prototype.slice.call(arguments));
-        if (this instanceof arguments.callee) {
+        if (this instanceof f) {
```
Comment (spicyj) [inline src/test/mocks.js]: (These were necessary because babel compiles everything in strict mode.)

**Comment by zpao** [comment]
This will (potentially) break with closure compiler. It might not (they might already be in the DOM externs) but we should confirm

```
-  'Esc': 'Escape',
-  'Spacebar': ' ',
-  'Left': 'ArrowLeft',
-  'Up': 'ArrowUp',
-  'Right': 'ArrowRight',
-  'Down': 'ArrowDown',
-  'Del': 'Delete',
-  'Win': 'OS',
-  'Menu': 'ContextMenu',
-  'Apps': 'ContextMenu',
-  'Scroll': 'ScrollLock',
-  'MozPrintableKey': 'Unidentified'
+  Esc: 'Escape',
+  Spacebar: ' ',
+  Left: 'ArrowLeft',
+  Up: 'ArrowUp',
+  Right: 'ArrowRight',
+  Down: 'ArrowDown',
+  Del: 'Delete',
+  Win: 'OS',
+  Menu: 'ContextMenu',
+  Apps: 'ContextMenu',
+  Scroll: 'ScrollLock',
+  MozPrintableKey: 'Unidentified'

```

**Comment by spicyj** [comment]
I guess so. Filed https://github.com/babel/babel/issues/1415 upstream.

**Comment by sebmck** [comment]
Sorry for the delay. Released Babel 5.2.7 which doesn't do that by default anymore.

**Comment by chicoxyzzy** [comment]
oh yeah! make it!

**Comment by spicyj** [comment]
@zpao r? Updated babel to avoid property key minifiers.

**Comment by zpao** [comment]
r=zpao

**Comment by zpao** [comment]
Just remembered that we can probably get rid of our es3ify dependency now (which means a bit of cleanup in the browserify config)

**Comment by thejameskyle** [comment]
:tada:

**Comment by sebmck** [comment]
yesss

**Pull request merged**